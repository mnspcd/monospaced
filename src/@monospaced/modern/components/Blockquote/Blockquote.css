.mdrn .Blockquote {
  margin-bottom: 0;
  margin-left: 1em;
  margin-right: 0;
  margin-top: 0;
  max-width: calc(var(--measure) -1em);
}

.mdrn .Blockquote p {
  margin-bottom: 0;
  margin-top: 0;
  quotes: "“" "”" "‘" "’";
}

.mdrn .Blockquote p:first-child::before {
  content: open-quote;
  margin-left: -0.75ch;
}

.mdrn .Blockquote p:first-child::after {
  content: close-quote;
}

.mdrn .Blockquote p + p:last-child {
  text-align: right;
}

@media (--larger-viewport) {
  .mdrn .Blockquote p + p:last-child {
    font-size: calc(1em * var(--type-scale-s));
    line-height: calc(var(--leading) / var(--type-scale-s));
    max-width: 100%;
  }
}
