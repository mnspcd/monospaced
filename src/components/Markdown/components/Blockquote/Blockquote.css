.Blockquote {
  margin-bottom: 0;
  margin-left: 1em;
  margin-right: 0;
  margin-top: 0;
  max-width: calc(var(--measure) -1em);
}

.Blockquote p {
  margin-bottom: 0;
  margin-top: 0;
  quotes: "“" "”" "‘" "’";
}

.Blockquote p:first-child::before {
 content: open-quote;
 margin-left: -0.75ch;
}

.Blockquote p:first-child::after {
 content: close-quote;
}

.Blockquote p + p:last-child {
  text-align: right;
}

@media (--larger-viewport) {
  .Blockquote p + p:last-child {
    font-size: calc(1em * var(--type-scale-s));
    line-height: calc(var(--leading) / var(--type-scale-s));
    max-width: 100%;
  }
}
