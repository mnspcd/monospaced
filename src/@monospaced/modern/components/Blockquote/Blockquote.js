import PropTypes from "prop-types";
import React from "react";

import "./Blockquote.css";

/**
 * Use `Blockquote` to typeset quotations and their attribution.
 */
const Blockquote = ({ children }) => (
  <blockquote className="Blockquote">{children}</blockquote>
);

Blockquote.propTypes = {
  /**
   * Text to display in the Blockquote. Usually an HTML paragraph element
   * containing a quotation. With an optional second paragraph containing
   * its attribution.
   */
  children: PropTypes.node.isRequired,
};

export default Blockquote;
