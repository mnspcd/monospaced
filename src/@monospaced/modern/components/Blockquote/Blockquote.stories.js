import { text } from "@storybook/addon-knobs";
import React from "react";

import Blockquote from "./Blockquote";

export default {
  component: Blockquote,
  title: "Components/Blockquote",
};

export const basic = () => (
  <Blockquote>
    <p>
      One more attribute the modern typographer must have: the capacity for
      taking great pains with seemingly unimportant detail. To him, one
      typographical point must be as important as one inch, and he must harden
      his heart against the accusation of being too fussy.
    </p>
    <p>— Hans P. Schmoller, in ‘Book Design Today’, Spring 1951</p>
  </Blockquote>
);

export const knobs = () => {
  const quote = text(
    "quote",
    `One more attribute the modern typographer must have: the capacity for taking great pains with seemingly unimportant detail. To him, one typographical point must be as important as one inch, and he must harden his heart against the accusation of being too fussy.`,
  );
  const source = text(
    "source",
    "— Hans P. Schmoller, in ‘Book Design Today’, Spring 1951",
  );

  return (
    <Blockquote>
      {quote && <p>{quote}</p>}
      {quote && source && <p>{source}</p>}
    </Blockquote>
  );
};

knobs.story = { parameters: { docs: { disable: true } } };
