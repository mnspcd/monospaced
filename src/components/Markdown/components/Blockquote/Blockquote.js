import PropTypes from "prop-types";
import React from "react";

import "./Blockquote.css";

const Blockquote = ({ children }) => (
  <blockquote className="Blockquote">{children}</blockquote>
);

Blockquote.propTypes = { children: PropTypes.node.isRequired };

export default Blockquote;
