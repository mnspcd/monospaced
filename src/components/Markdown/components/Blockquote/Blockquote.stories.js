import React from "react";
import { storiesOf } from "@storybook/react";
import { text } from "@storybook/addon-knobs";

import Blockquote from "./Blockquote";

storiesOf("Components/Markdown/Blockquote", module).add("Component", () => {
  const quote = text(
    "quote",
    `One more attribute the modern typographer must have: the capacity for taking great pains with seemingly unimportant detail. To him, one typographical point must be as important as one inch, and he must harden his heart against the accusation of being too fussy.`,
  );
  const source = text(
    "source",
    "— Hans P. Schmoller, in ‘Book Design Today’, Spring 1951",
  );

  return (
    <Blockquote>
      {quote && <p>{quote}</p>}
      {quote && source && <p>{source}</p>}
    </Blockquote>
  );
});
