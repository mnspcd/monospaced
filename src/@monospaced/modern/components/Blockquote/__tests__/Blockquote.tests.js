/* eslint-env jest */
import { render } from "@testing-library/react";
import React from "react";

import Blockquote from "..";

describe("Blockquote component", () => {
  it("should render correctly", () => {
    const { container } = render(<Blockquote>children</Blockquote>);

    expect(container).toMatchSnapshot();
  });
});
