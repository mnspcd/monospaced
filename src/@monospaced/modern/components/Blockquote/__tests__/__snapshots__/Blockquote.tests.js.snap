// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote component should render correctly 1`] = `
<div>
  <blockquote
    class="Blockquote"
  >
    children
  </blockquote>
</div>
`;
